controllers/grid/languages/form/AddLanguageForm.php

```diff
@@ -0,0 +1,139 @@
+<?php
+
+/**
+ * @file controllers/grid/languages/form/AddLanguageForm.php
+ *
+ * Copyright (c) 2014-2023 Simon Fraser University
+ * Copyright (c) 2003-2023 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class AddLanguageForm
+ *
+ * @ingroup controllers_grid_languages_form
+ *
+ * @brief Form for adding languages.
+ */
+
+namespace PKP\controllers\grid\languages\form;
+
+use APP\core\Application;
+use APP\core\Services;
+use APP\template\TemplateManager;
+use PKP\db\DAORegistry;
+use PKP\facades\Locale;
+use PKP\form\Form;
+use PKP\i18n\LocaleMetadata;
+use PKP\site\SiteDAO;
+
+class AddLanguageForm extends Form
+{
+    /**
+     * Constructor.
+     */
+    public function __construct()
+    {
+        parent::__construct('controllers/grid/languages/addLanguageForm.tpl');
+    }
+
+    //
+    // Overridden methods from Form.
+    //
+    /**
+     * @copydoc Form::initData()
+     */
+    public function initData()
+    {
+        parent::initData();
+        $this->setData('addedLocales', ((Application::get()->getRequest())->getContext())->getSupportedAddedSubmissionLocales());
```

Comment:
At the beginning getSupportedAddedSubmissionLocales() is null, which needs to be considered here, I think.